### DISCUSSION TITLE
CRPSS missing in EnsembleStat

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#1740] [Configuration] [answered]
Again, refer to the issue#1725, I just found that following setting only works for the case in which fcst, obs and climo data
have same field names / levels: 

ENSEMBLE_STAT_CLIMO_MEAN_FILE_NAME = {climpath}/mean_{valid?fmt=%m%d}
ENSEMBLE_STAT_CLIMO_MEAN_USE_FCST = True
ENSEMBLE_STAT_CLIMO_STDEV_FILE_NAME = {climpath}/stdev_{valid?fmt=%m%d}
ENSEMBLE_STAT_CLIMO_STDEV_USE_FCST = True

But if  the field names in fcst or obs are different from the same fields in climo data, the above
setting is not working. For example, for ECMWF ensemble, the fcst and obs have same field names.
but climo data have different field names from the those in fcts and obs data.

I tested this case in MET outside of METplus. The CRPSS can be created from EnsembleStat.
The ens, fcst, obs and climo settings in MET config looks are as following
ens = {
   ens_thresh = 1.0;
   vld_thresh = 1.0;

   field = [

      {
         name       = "GH";
         level      = "P500";
         GRIB1_ptv = 128;
         GRIB1_center = 98;
         set_attr_name = "HGT";
       }
   ];
}
fcst = ens
obs = fcst

climo_mean = fcst; 
climo_mean = {

   file_name = "${CLIM_MEAN}";

   field = [

      {
         name       = "HGT";
         level      = "P500";
      }
  ];

}

The issue #1725 has been closed, so I open this new issue.

Thanks!

Binbin 
EMC/NCEP

**Comment by georgemccabe** [comment, marked as answer]
@BinbinZhou-NOAA,
If the field name for the climo field does not match the fcst or obs, then you would not set `ENSEMBLE_STAT_CLIMO_MEAN_USE_FCST` and instead set:
`ENSEMBLE_STAT_CLIMO_MEAN_FIELD = { name="HGT"; level="P500"; }`

The ENSEMBLE_STAT_CLIMO_MEAN_USE_FCST variable adds `climo_mean = fcst;` which essentially just obtains the field list from the fcst dictionary. If the fields are not the same, then you don't need to set this.

Let me know if this does not resolve your issue.

**Reply by georgemccabe** [reply]
Also, if you are using METplus 5.0.0-beta1 or higher, you can also set the climo field info like this:

```
ENSEMBLE_STAT_CLIMO_MEAN_VAR1_NAME = HGT
ENSEMBLE_STAT_CLIMO_MEAN_VAR1_LEVELS = P500
```

**Reply by BinbinZhou-NOAA** [reply]
I tried this, but not resolves the issue

My setting now is
ENSEMBLE_STAT_CLIMO_MEAN_FILE_NAME = {climpath}/mean_{valid?fmt=%m%d}
ENSEMBLE_STAT_CLIMO_MEAN_USE_FCST = climo_mean = fcst;

ENSEMBLE_STAT_CLIMO_MEAN_FIELD = { name="HGT"; level="P500"; }, { name="UGRD"; level="P850"; }, { name="VGRD"; level="P850"; }, { name="TMP"; level="P850"; }, { name="UGRD"; level="Z10"; }, { name="VGRD"; level="Z10"; }, { name="PRMSL"; level="L0"; }, { name="HGT"; level="P700"; }, { name="HGT"; level="P1000"; }, { name="TMP"; level="P500"; }, { name="UGRD"; level="P250"; }, { name="VGRD"; level="P250"; }

**Reply by BinbinZhou-NOAA** [reply]
If remove ENSEMBLE_STAT_CLIMO_MEAN_USE_FCST = climo_mean = fcst;
Still not working

**Reply by georgemccabe** [reply]
@BinbinZhou-NOAA, could you send an updated log file?

**Reply by BinbinZhou-NOAA** [reply]
I have sent entire running directory to HERA 
/scratch2/NCEPDEV/fv3-cam/noscrub/Binbin.Zhou/evs_atmos_global_ens_ecme_grid2grid_stats.134739
The output_base is in its sub-directory  grid2grid/run_ecme_valid_at_t00z_fhr1
Final conf. logs and stat files can be found inside

**Reply by georgemccabe** [reply]
Thank you for sending the new files. It looks like you are setting the climo_mean, but not the climo_stdev. You will also need to set ENSEMBLE_STAT_CLIMO_STDEV_FIELD and ENSEMBLE_STAT_CLIMO_STDEV_FILE_NAME to the appropriate values to compute the CRPS score.

**Reply by BinbinZhou-NOAA** [reply]
After set both ENSEMBLE_STAT_CLIMO_STDEV_FIELD and ENSEMBLE_STAT_CLIMO_STDEV_FILE_NAME, now CRPSS can be generated!
Thanks!
